src/rest/SearchEndpoint.ts

```diff
@@ -873,4 +821,404 @@ export class SearchEndpoint implements ISearchEndpoint {
+          method: '',
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Add the accessToken to the query string arguments
+ */
+function accessTokenInUrl(){
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    let originalMethod = descriptor.value;
+    let nbParams = target[key].prototype.constructor.length;
+
+    descriptor.value = function(...args: any[]) {
+      let queryString = this.buildAccessToken();
+      if(args[nbParams-1]){
+        args[nbParams-1].queryString = args[nbParams-1].queryString.concat(queryString);
+      }
+      else{
+        let params: IEndpointCallParameters = {
+          url: '',
+          queryString: queryString,
+          requestData: {},
+          method: '',
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Add the search object to the query string arguments
+ * @param fromOptions Take the search object from the callOptions instead of the first parameter of the query
+ * Note : This tag requires the method's first parameter to be the search object as an ISearch if fromOptions is set to false
+ */
+function searchObjectInUrl(fromOptions: boolean = false){
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    let originalMethod = descriptor.value;
+    let nbParams = target[key].prototype.constructor.length;
+
+    descriptor.value = function(...args: any[]) {
+      let queryString;
+      if(fromOptions){
+        let callOptions = _.extend({}, args[nbParams-2]);
+        queryString = this.buildCompleteQueryString(callOptions.query, callOptions.queryObject);
+      }
+      else{
+        queryString = this.buildCompleteQueryString(null, args[0]);
+      }
+      if(args[nbParams-1]){
+        args[nbParams-1].queryString = args[nbParams-1].queryString.concat(queryString);
+      }
+      else{
+        let params: IEndpointCallParameters = {
+          url: '',
+          queryString: queryString,
+          requestData: {},
+          method: '',
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Add data into que body of the request
+ * @param fromOptions Take the search object from the callOptions instead of the first parameter of the query
+ * Note : This tag requires the method's first parameter to be the data to add if fromOptions is set to true
+ */
+function requestData(fromOptions: boolean = false){

```

Comment:
Hum, after thinking about this, this decorator does not make much sense.

Any call sends data will use this decorator : This is redundant. I would move back the logic inside each method.

Each method job will be to populate to perform requestData manually, essentially.